src/app/routes/onDemandRadio/getInitialData/pathWithLogging/index.js

```diff
@@ -0,0 +1,32 @@
+import isNil from 'ramda/src/isNil';
+import path from 'ramda/src/path';
+import { RADIO_FIELD_MISSING } from '#lib/logger.const';
+import { getUri } from '../logInitialData';
+import nodeLogger from '#lib/logger.node';
+
+const logger = nodeLogger(__filename);
+
+export const LOG_LEVELS = {
+  INFO: 'INFO',
+  WARN: 'WARN',
+  ERROR: 'ERROR',
+};
+
+const getLogger = level =>
+  ({
+    [LOG_LEVELS.INFO]: logger.info,
+    [LOG_LEVELS.WARN]: logger.warn,
+    [LOG_LEVELS.ERROR]: logger.error,
+  }[level] || logger.info);
+
+export default (fieldPath, { logLevel = LOG_LEVELS.INFO } = {}) => pageData => {
+  const value = path(fieldPath, pageData);
+  if (isNil(value)) {
+    getLogger(logLevel)(RADIO_FIELD_MISSING, {
```

Comment:
I sometimes get confused by this, and wondered if this would make it a bit more readable?
```suggestion
    const fileLogger =  getLogger(logLevel);
    fileLogger(RADIO_FIELD_MISSING, {
```

Reply:
👍